src/opm/parser/eclipse/EclipseState/Schedule/Schedule.cpp

```diff
@@ -2824,8 +2795,15 @@ void Schedule::invalidNamePattern( const std::string& namePattern,  std::size_t
         for (const auto& wname : well_names) {
             const auto& well = this->getWell(wname, timeStep);
             const auto& connections = well.getConnections();
-            if (connections.allConnectionsShut())
-                this->updateWellStatus( well.name(), timeStep, Well::Status::SHUT, false);
+            if (connections.allConnectionsShut()) {
+                if (well.getStatus() != Well::Status::SHUT) {
```

Comment:
If you wanted to, you could fuse these two conditions into one to save one level of nesting.

Reply:
I have updated according to the suggestion.